tool/src/org/antlr/v4/codegen/model/InvokeRule.java

```diff
@@ -24,40 +24,43 @@
 
 /** */
 public class InvokeRule extends RuleElement implements LabeledOp {
-	public String name;
-	public OrderedHashSet<Decl> labels = new OrderedHashSet<Decl>(); // TODO: should need just 1
-	public String ctxName;
+	public final String originalName;
+	public final String name;
```

Comment:
shouldn't this be the original name?  what is the difference? if it is the escaped name then we should probably call it that.

Reply:
`originalName` is used in templates for expression with concatenation or with case changing: `<r.originalName>_sempred`, `<f.originalName; format="cap">`. Just `name` is used for plain identifiers: `<r.name>(...)`.